### PR TITLE
Add TimeTeller

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
This can replace the TimeTeller interface and implementations in the FundraisingFrontend

Not 100% happy with the naming but can't think of anything better

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Also in case you are wondering "why clock implementation 4358967396?": I found some sane implementations but none of those work with PHP 5.6.

File: README.md [hunk 1]
```diff
@@ -4,18 +4,34 @@
 
 Simple interface to get the current time without binding to global system resources. PHP 5.6+
 
-## Usage
+## Clock Usage
 
 ```php
 function yourCode(Clock $clock) {
-	$time = $clock->now(); // No global access and easily testable
+    $time = $clock->now(); // Returns DateTimeImmutable. No global access and easily testable
```
Comment (gbirke) [inline README.md]: This comment is essential. When I looked at the Readme of the master branch, I wondered what the return type of `now` would be.

**Comment by JeroenDeDauw** [comment]
I see what you are saying and am not sure having TimeTeller makes sense. However note that TimeTeller is not the same as a ClockFormatter. From the consumer PoV there is no notion of Clock. The consumer just knows it gets a string representing the current time. That can be a better interface than getting a DateTimeImmutable and calling format on it with some format string. I vaguely recall that I thought the formatting responsibility in the FF code was misplaced before the extraction into TimeTeller happened. Thoughts?

**Comment by JeroenDeDauw** [comment]
Trimmed down version of the code using TimeTeller:

```php
class JsonBucketLogger implements BucketLogger {

	private $logWriter;
	private $timeTeller;

	public function __construct( LogWriter $logWriter, TimeTeller $timeTeller ) {
		$this->logWriter = $logWriter;
		$this->timeTeller = $timeTeller;
	}

	private function formatData( LoggingEvent $event, Bucket ...$buckets ): array {
		return [
			'date' => $this->timeTeller->getTime(),
			'eventName' => 'stuff'
		];
	}

}
```

This class uses to have a $dateFormat parameter. If the format needs to be configurable, using the TimeTeller interface has the advantage of fully removing the formatting responsibility from JsonBucketLogger. You won't need to test that it uses the formatting function correctly. On the other hand you do add an extra thing to test for integration tests, though they only need to test the correct TimeTeller is used.

I'm not sure this advantage is worth it. And for the users of the library that don't need TimeTeller, having one interface will be more clear than two.

**Comment by gbirke** [comment]
Thanks for not including the TimeTeller. The date format of JsonBucketLogger was only configurable for tests.

**Comment by JeroenDeDauw** [comment]
Huh GitHub... I did not merge THIS pr! Such confoosing.

**Pull request merged**